dbt Core quickstart with GH codespaces (#3059)
fishtown-analytics/docs.getdbt.com: website/docs/docs/quickstarts/dbt-core/codespace-qs.md

---
title: "Quickstart for dbt Core using GitHub Codespaces"
id: codespace
description: "Quickly get set up to start running dbt commands in a codespace environment."
sidebar_label: "GitHub Codespace quickstart"
---

In this quickstart guide, you’ll learn how to create a codespace and be able to execute the `dbt build` command from it in _less than 5 minutes_. 

> dbt Labs provides a [GitHub Codespace](https://docs.github.com/en/codespaces/overview) template that you (and anyone else) can reuse to create a complete dbt environment with a working and runnable project. When you create the codespace, the [dev container](https://docs.github.com/en/codespaces/setting-up-your-project-for-codespaces/adding-a-dev-container-configuration/introduction-to-dev-containers) creates a fully functioning dbt environment, connects to a DuckDB database, and loads a year of fictional Jaffle Shop data into the database for you. Jaffle Shop is a fictional cafe selling food and beverages in several US cities.

Reviewer: Avoiding some redundant language

```suggestion
dbt Labs provides a [GitHub Codespace](https://docs.github.com/en/codespaces/overview) template that you (and anyone else) can reuse to create a complete dbt environment with a working and runnable project. When you create the codespace, the [dev container](https://docs.github.com/en/codespaces/setting-up-your-project-for-codespaces/adding-a-dev-container-configuration/introduction-to-dev-containers) creates a fully functioning dbt environment, connects to a DuckDB database, and loads a year of data from our fictional Jaffle Shop café, which sells food and beverages in several US cities. 
```

> :::tip Check out our Jaffle Shop template 

Reviewer: Since we're trying to reduce callouts would this make sense as a note at the end of the previous paragraph?

The [README](https://github.com/dbt-labs/jaffle-shop-template#readme) for the Jaffle Shop template also provides instructions on how to do this, along with animated GIFs. You can check those out, too.  

:::

## Prerequisites

> - To use the dbt CLI, it's important that you know some basics of the terminal. In particular, you should understand `cd`, `ls` , and `pwd` to navigate through the directory structure of your computer easily.

Reviewer: Since this is for beginners:

```suggestion
- To use the dbt command-line interface (CLI), it's important that you know some basics of the terminal. In particular, you should understand `cd`, `ls` , and `pwd` to navigate through the directory structure of your computer easily.
```

- You have a [GitHub account](https://github.com/join).

## Create a codespace

1. Go to the `jaffle-shop-template` [repository](https://github.com/dbt-labs/jaffle-shop-template) after you log in to your GitHub account. 
1. Click **Use this template** at the top of the page and choose **Create new repository**.
1. Click **Create repository from template** when you’re done setting the options for your new repository.

> 1. Click **Code** (at the top of the new repository’s page). Under the **Codespaces** tab,  choose **Create codespace on main**. Depending on how you've configured your computer's setting, this either opens a new browser tab with VSCode running in it or opens a new VSCode window with the codespace in it. 

Reviewer: Adding this for clarity since running these steps doesn't produce any apparent VSCode branding on the screen except for a brief line that goes away automatically and is easily missed
```suggestion
1. Click **Code** (at the top of the new repository’s page). Under the **Codespaces** tab,  choose **Create codespace on main**. Depending on how you've configured your computer's settings, this either opens a new browser tab with the Codespace development environment with VSCode running in it or opens a new VSCode window with the codespace in it. 
```

Reviewer: Will this only work with VSCode if they open it via app or will it work with any (most?) code editors?

Author: looks like GH Codespace supports other editors  https://github.com/features/codespaces depending on how the dev container is configured

1. Wait for the codespace to finish building by waiting for the `postCreateCommand` command to complete; this can take several minutes:

    <Lightbox src="/img/codespace-quickstart/postCreateCommand.png" title="Wait for postCreateCommand to complete" />

    When this command completes, you can start using the codespace development environment. The terminal the command ran in will close and you will get a prompt in a brand new terminal. 

> 

Reviewer: For all of these instructions below will they follow the exact same workflows regardless of whether they're working from the browser or the app with command line? Are there limitations on the Codespace browser tools they need to be aware of?

    
    For complete information, refer to the [dbt command reference](https://docs.getdbt.com/reference/dbt-commands). Common commands are:
    
    - [dbt compile](https://docs.getdbt.com/reference/commands/compile) — generates executable SQL from your project source files
    - [dbt run](https://docs.getdbt.com/reference/commands/run) — compiles and runs your project
    - [dbt test](https://docs.getdbt.com/reference/commands/test) — compiles and tests your project
    - [dbt build](https://docs.getdbt.com/reference/commands/build) — compiles, runs, and tests your project

## Generate a larger data set

If you'd like to work with a larger selection of Jaffle Shop data, you can generate an arbitrary number of years of fictitious data from within your codespace. 

1. Install the Python package called [jafgen](https://pypi.org/project/jafgen/). At the terminal's prompt, run:

    ```shell
    /workspaces/test (main) $ pip install jafgen
    ```

1. When installation is done, run:
    ```shell
    /workspaces/test (main) $ jafgen --years NUMBER_OF_YEARS
    ``` 

>     Replace `NUMBER_OF_YEARS` with the number of years you want to simulate. This command builds the CSV files and stores them in the `jaffle-data` folder, and is automatically sourced based on the `sources.yml` file and the `dbt-duckdb` adapter.

Reviewer 2: I feel like we should credit Josh / the `dbt-duckdb` adapter here in some way. Perhaps by linking here to `reference/warehouse-setups/duckdb-setup`?

> As you increase the number of years, it takes exponentially more time to generate the data because the Jaffle Shop stores grow in size and number. For a good balance of data size and time to build, dbt Labs suggests a maximum of 6 years.

Reviewer: This is a super awesome feature. I didn't know we could do this with that data.

## Related content
- [Create a GitHub repository](/docs/quickstarts/dbt-core/manual-install#create-a-repository)
- [Build your first models](/docs/quickstarts/dbt-core/manual-install#build-your-first-models)
- [Test and document your project](/docs/quickstarts/dbt-core/manual-install#test-and-document-your-project)

> - [Schedule a job](/docs/quickstarts/dbt-core/manual-install#schedule-a-job)

Author: are the 4 links above appropriate? 

is the list missing anything (like `/docs/quickstarts/dbt-core/manual-install#create-a-project`)? i'm thinking no  because of the codespace env but want to double-check with you.
